app/model.py

```diff
@@ -28,3 +28,30 @@ def sort_mentors(mentor: Dict) -> Tuple:
             key=sort_mentors,
         )[:n_matches]
         return [mentor["profile_id"] for mentor in results]
+
+
+class MentorMatcherSearch:
+    """Callable matching class implementing sorted search algorithm."""
+    db = MongoDB()
+
+    def __call__(self, profile_id: str, n_matches: Optional[int] = None) -> List[str]:
+        """Return a list of profile_id for matched mentors."""
+        mentor = self.db.first("Mentors", {"profile_id": profile_id})
+
+        def sort_mentees(mentee: Dict) -> Tuple:
+            return (
+                mentor["pair_programming"] != mentee["pair_programming"],
+                mentor["job_help"] != mentee["job_help"],
+                not mentee["industry_knowledge"],
+            )
+
+        results = sorted(
+            self.db.read("Mentees", {
+                "tech_stack": mentor["tech_stack"],
+                "is_active": True,
+                "accepting_new_mentors": True,
+                "validate_status": 'approved'
```

Comment:
Verify that mentees have the following fields: `is_active`, `accepting_new_mentors` & `validate_status`. If so, double check that the data types match what we have on the mentor side.

Reply:
Looks like we should remove line 51, it refers to a field that does not exist: `accepting_new_mentors`.